### ISSUE TITLE
Provide a way to completely turn off the cache (bring back disableCache)

### ISSUE THREAD

**Issue by tmartensen** [wro4j/wro4j#957]
We're trying to upgrade to 1.7.8 due to a bug in 1.7.3 that prevents `minimizeEnabled` from working. When we upgraded, however, we've found that `disableCache` is now gone, and setting these values seem to have no effect on invalidating the cache when a resource is modified:
```
resourceWatcherUpdatePeriod=5
resourceWatcherAsync=true
cacheUpdatePeriod=5
modelUpdatePeriod=5

gzipResources=false
parallelPreprocessing=true
connectionTimeout=2000
encoding=UTF-8
ignoreEmptyGroup=true
ignoreFailingProcessor=false
cacheGzippedContent=false
jmxEnabled=true
managerFactoryClassName=ro.isdc.wro.manager.factory.ConfigurableWroManagerFactory
preProcessors=cssImport,semicolonAppender,cssMin
postProcessors=cssVariables,jsMin
uriLocators=servletContext,uri,classpath
```

Also, it seems that that refreshing the browser while the cache is being dumped sometimes catches the concatenated JavaScript in an unfinished state, leading to inconsistent results and random JavaScript errors.

It would seem that for development purposes, a way to completely disable the cache (such as the `disableCache` flag) would be preferable to setting timeouts that may not work for rapid development (<5 second) cycles. Thoughts?

**Comment by alexo** [comment]
@tmartensen I'm not sure why using short lived cacheUpdatePeriod doesn't work for your use-case?

**Comment by tmartensen** [comment]
Having a dummy CacheStrategy that could be turned on via wro.properties would be great!

Our project is using Angular.js,jQuery, and a few other libraries, and is currently around 70,000 lines long when it is unminified, which is how it is served when working locally. I'm guessing that the file isn't done being processed by the time that Angular tries to load the modules, which is why it breaks. In order to reproduce consistently, all I have to do is refresh the browser twice in rapid succession. The project I'm working on is owned by a client, but I'll try to put together a small project and see if I can reproduce.

**Comment by dacofr** [comment]
How to define a CacheStrategy implementation in the wro.properties ? (currently using 1.7.6)
I do not find in the documentation https://code.google.com/p/wro4j/wiki/ConfigurationOptions

I'm agree with @tmartensen, a solution to disable cache will be more efficience in developpement mode than checking if a ressources has changed

Thanks

**Comment by alexo** [comment]
In order to configure a different cacheStrategy, you can add the following
config in wro.properties (assuming ConfigurableWroManagerFactory is used):

cacheStrategy=myCustomStrategy

the "*myCustomStrategy*" is an alias used to identify a CacheStrategy
implementation provided as an SPI.

Alternatively, it is possible to create custom WroManagerFactory and set
the cacheStrategy explicitly in code.

Cheers,
Alex

**Comment by dacofr** [comment]
Great I already have a custom WroManagerFactory.

Thanks.

Daco

**Comment by dacofr** [comment]
In my case, works like a charm with :

a wro.properties in WEB-INF/
```
managerFactoryClassName=fr.planetvo.util.web.wro.PvoConfigurableWroManagerFactory
```

My own WroManagerFactory
```java
public class PvoConfigurableWroManagerFactory extends ConfigurableWroManagerFactory
{
	@Override
	protected CacheStrategy<CacheKey, CacheValue> newCacheStrategy()
	{
		return new NoCacheStrategy<>();
	}
}
```

And my NoCacheStrategy
```java
public class NoCacheStrategy<K, V>
	implements CacheStrategy<K, V>
{
	@Override
	public void put(final K k, final V v)
	{ // Noting to do
	}

	@Override
	public V get(final K k)
	{
		return null;
	}

	@Override
	public void clear()
	{ // Noting to do 
	}

	@Override
	public void destroy()
	{ // Noting to do
	}
}
```


I've try with this in my wro.properties
```
cachingStrategy=nocache
```
but i don't know how to modify override ro.isdc.wro.cache.spi.DefaultCacheStrategyProvider to provide my alias like  
```
map.put("nocache", new NoCacheStrategy<CacheKey, CacheValue>());
```

I you agree, i can try a PR to provide a no cache strategy ?

**Comment by alexo** [comment]
@dacofr Sure, it would be great. Thanks!

**Comment by dacofr** [comment]
PR #979 submited

**Comment by alexo** [comment]
@dacofr  Thanks!

**Comment by dacofr** [comment]
No problem :)

Maybe an idea on wro4j 1.7.9 release ?

**Comment by alexo** [comment]
I was planning to fix few more issues before releasing it... but probably it is a good idea to release with whatever is already fixed.

**Comment by alexo** [comment]
@dacofr a new released has been promoted to maven central repo and it should be available in a couple of hours. I will announce the release later. Thank you for your contribution.

**Comment by Bolerr** [comment]
Was this released as part of 1.7.9?

**Comment by alexo** [comment]
@Bolerr yes. The release notes are available here: https://github.com/wro4j/wro4j/issues?q=milestone%3A1.7.9+is%3Aclosed
